File: src/main/java/com/gmail/borlandlp/minigamesdtools/events/ActivePointDestroyedEvent.java
package com.gmail.borlandlp.minigamesdtools.events;

import com.gmail.borlandlp.minigamesdtools.activepoints.ActivePoint;
import org.bukkit.entity.Player;
import org.bukkit.event.Cancellable;
import org.bukkit.event.Event;
import org.bukkit.event.HandlerList;

public class ActivePointDestroyedEvent extends Event implements Cancellable {
    private ActivePoint activePoint;
    private Player destroyer;
    private boolean canceled;
    private static final HandlerList handlers = new HandlerList();

    public ActivePointDestroyedEvent(ActivePoint activePoint1, Player destroyer1) {
        this.activePoint = activePoint1;
        this.destroyer = destroyer1;
    }

    public ActivePoint getActivePoint() {
        return activePoint;
    }

    public Player getDestroyer() {
        return destroyer;
    }

    @Override
    public boolean isCancelled() {
        return this.canceled;
    }

    @Override
    public void setCancelled(boolean b) {
        this.canceled = b;
    }

    @Override
    public HandlerList getHandlers()
    {
        return handlers;
    }

    public static HandlerList getHandlerList() {
        return handlers;
    }
}

File: src/main/kotlin/com/gmail/borlandlp/minigamesdtools/events/ActivePointDestroyedEvent.kt
package com.gmail.borlandlp.minigamesdtools.events

import com.gmail.borlandlp.minigamesdtools.activepoints.ActivePoint
import org.bukkit.entity.Player
import org.bukkit.event.Cancellable
import org.bukkit.event.Event
import org.bukkit.event.HandlerList

class ActivePointDestroyedEvent(val activePoint: ActivePoint, val destroyer: Player) : Event(), Cancellable {
    private var canceled = false

    override fun isCancelled(): Boolean {
        return canceled
    }

    override fun setCancelled(b: Boolean) {
        canceled = b
    }

    override fun getHandlers(): HandlerList {
        return handlerList
    }

    companion object {
        @JvmStatic
        val handlerList = HandlerList()
    }
}